### ISSUE TITLE
`isCoveredNode` false positive

### ISSUE THREAD

**Issue by RunDevelopment** [ota-meshi/eslint-plugin-regexp#178]
I found a false positive the `isCoveredNode`method produces.

```js
  /(^|\s)[+-]?(?:\d*\.\d+|\d+\.\d*|\d+)(?:e[+-]?\d+)?(?=\s|$)/i
//               ^~~~~~~~ ^~~~~~~~
//                   ^       / 
//                    \_____/ is covered by
```

`isCoveredNode` thinks that `\d+\.\d*` (2nd alt) is covered by `\d*\.\d+` (1st alt).

I found this bug because of a false positive the `no-dupe-disjunction` rule produced because of it.
